front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -0,0 +1,207 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { keyBy, sortBy } from 'lodash';
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi, type TrainScheduleResult } from 'common/api/osrdEditoastApi';
+import useInfraStatus from 'modules/pathfinding/hooks/useInfraStatus';
+import { setFailure } from 'reducers/main';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+import useLazyLoadTrains from './useLazyLoadTrains';
+import useScenario from './useScenario';
+import useSimulationResults from './useSimulationResults';
+import selectTrainAndPathForProjection from '../helpers/selectTrainAndPathForProjection';
+
+const useScenarioData = () => {
+  const dispatch = useAppDispatch();
+  const trainIdUsedForProjectionInStore = useSelector(getTrainIdUsedForProjection);
+  const selectedTrainIdInStore = useSelector(getSelectedTrainId);
+
+  const [trainSchedules, setTrainSchedules] = useState<TrainScheduleResult[]>();
+  const [trainIdUsedForProjection, setTrainIdUsedForProjection] = useState<number>();
+  const [selectedTrainId, setSelectedTrainId] = useState<number>();
+  const [trainIdsToFetch, setTrainIdsToFetch] = useState<number[]>();
+  const [trainIdsToRemove, setTrainIdsToRemove] = useState<number[]>([]);
+
+  const scenario = useScenario();
+
+  const { infra, isInfraLoaded, reloadCount } = useInfraStatus();
+
+  const { data: timetable } = osrdEditoastApi.endpoints.getV2TimetableById.useQuery(
+    { id: scenario?.timetable_id as number },
```

Comment:
Nit: I'd prefer `scenario?.timetable_id!` instead of a cast, to allow TypeScript to check that `timetable_id` is indeed a `number`.

Reply:
Done

Reply:
Seems like this has been lost at some point as well.

Reply:
Done again ✅